src/app/systelab-components/datepicker/datepicker.component.ts

```diff
@@ -237,7 +242,30 @@ export class Datepicker implements OnInit, AfterViewInit, DoCheck, OnDestroy {
 		}
 	}
 
-	public onKeyDown(event: KeyboardEvent) {
+	public formatDate(date: string ): string {
+		let dateTmp: string = date.trim();
+
+		if (dateTmp != null) {
+			if ((dateTmp.length === 8 || dateTmp.length === 10)
+				&& (dateTmp.lastIndexOf('/')  > 0 || dateTmp.lastIndexOf('-') > 0 || dateTmp.lastIndexOf('.') > 0)) {
+				dateTmp = dateTmp.replace('/', '')
+					.replace('-', '')
+					.replace('.', '');
+			}
+
+			if (dateTmp.length === 6) {
+				dateTmp = dateTmp.substring(0, 2) + '/' + dateTmp.substring(2, 4) + '/20'
+					+ dateTmp.substring(4, dateTmp.length);
+			} else if (dateTmp.length === 8) {
+				dateTmp = dateTmp.substring(0, 2) + '/' + dateTmp.substring(2, 4) + '/'
```

Comment:
Could you please group the different condicionals:
if ((dateTmp.length === 8 || dateTmp.length === 10) {
}
if (dateTmp.length === 6) {
} else if (dateTmp.length === 8) { ????????? A second time???????
}

Reply:
It can not be grouped because they are in different paths of the code.

Reply:
What happends if an user set the date with day and month only with one character? (1/1/2015)
The dateTmp.length will be 6, and this code will return 11/20/15

Reply:
This was used in Anthema for years, without problems.
However, and as this will be part of the systelab components, I will try to do it more robust.